### PR TITLE
Add rotation manipulator segment

### PR DESCRIPTION
**Pull request by hultonha**
## What does this PR do?

This PR adds a new feedback for a pairing of the `AngularManipulator` and `ManipulatorCircleView`. This gives much better feedback when interacting with the manipulator to see how far the entity has rotated.

## How was this PR tested?

This PR has been test primarily in the editor/viewport as this is a visual feature. It would be useful to follow-up with additional unit/integration tests and all existing manipulator tests will be run.

image

https://user-images.githubusercontent.com/82228511/183912110-278a919f-3dd6-456e-bab1-096494c30e74.mp4

Tagging @yuyihsu, @rainbj, @amzn-leenguy @bhanuja-s for visibility (tagging @shawstar, @benblack75 and @HogJonny-AMZN too).

### CODE REVIEW

**Review by AMZN-daimini — APPROVED** [review]

File: Code/Framework/AzToolsFramework/AzToolsFramework/ViewportSelection/EditorTransformComponentSelection.cpp [hunk 1]
```diff
@@ -3290,7 +3302,8 @@ namespace AzToolsFramework
         }
     }
 
-    void EditorTransformComponentSelection::OverrideComponentModeSwitcher(AZStd::shared_ptr<ComponentModeFramework::ComponentModeSwitcher> componentModeSwitcher)
+    void EditorTransformComponentSelection::OverrideComponentModeSwitcher(
+        AZStd::shared_ptr<ComponentModeFramework::ComponentModeSwitcher> componentModeSwitcher)
     {
         m_componentModeSwitcher = componentModeSwitcher;
     }
```
[the remark this answers is not in the document]
Comment (hultonha) [reply Code/Framework/AzToolsFramework/AzToolsFramework/ViewportSelection/EditorTransformComponentSelection.cpp:3309]: How so? 🤔 Isn't it right? I think it was incorrect before (line was too long)

[the remark this answers is not in the document]
Comment (AMZN-daimini) [reply Code/Framework/AzToolsFramework/AzToolsFramework/ViewportSelection/EditorTransformComponentSelection.cpp:3309]: Oh you're right, I guess I misinterpreted it when reviewing! Never mind :D

File: Code/Framework/AzToolsFramework/AzToolsFramework/ViewportSelection/EditorTransformComponentSelection.cpp [hunk 2]
```diff
@@ -3290,7 +3302,8 @@ namespace AzToolsFramework
         }
     }
 
-    void EditorTransformComponentSelection::OverrideComponentModeSwitcher(AZStd::shared_ptr<ComponentModeFramework::ComponentModeSwitcher> componentModeSwitcher)
+    void EditorTransformComponentSelection::OverrideComponentModeSwitcher(
+        AZStd::shared_ptr<ComponentModeFramework::ComponentModeSwitcher> componentModeSwitcher)
```
Comment (amzn-ahmadkrm) [inline Code/Framework/AzToolsFramework/AzToolsFramework/ViewportSelection/EditorTransformComponentSelection.cpp:3306]: nice find ;)

**Review by amzn-ahmadkrm — APPROVED** [review]
lgtm

File: Code/Framework/AzToolsFramework/AzToolsFramework/ViewportSelection/EditorTransformComponentSelection.cpp (hunk 2 above)
Comment (hultonha) [reply Code/Framework/AzToolsFramework/AzToolsFramework/ViewportSelection/EditorTransformComponentSelection.cpp:3306]: Haha it was just my good friend clang-format 🙂

File: Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/RotationManipulators.cpp [hunk 3]
```diff
@@ -135,8 +135,7 @@ namespace AzToolsFramework
 
         const float viewAlignedScale = 1.12f;
         m_viewAngularManipulator->SetView(CreateManipulatorViewCircle(
-            *m_viewAngularManipulator, AZ::Color(1.0f, 1.0f, 1.0f, 1.0f), radius * viewAlignedScale, m_circleBoundWidth,
-            DrawFullCircle));
+            *m_viewAngularManipulator, AZ::Color(1.0f, 1.0f, 1.0f, 1.0f), radius * viewAlignedScale, m_circleBoundWidth, DrawFullCircle));
```
Comment (moraaar) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/RotationManipulators.cpp:138]: nit: You can take advantage of already created colors: `AZ::Colors::White`.

**Review by moraaar — APPROVED** [review]

File: Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/RotationManipulators.cpp (hunk 3 above)
Comment (hultonha) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/RotationManipulators.cpp]: Good call, I can update this 👍

**Pull request merged**